install.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+set -e
+
+REPO="GoogleCloudPlatform/kubectl-ai"
+BINARY="kubectl-ai"
+
+# Detect OS
+OS="$(uname | tr '[:upper:]' '[:lower:]')"
+case "$OS" in
+  linux)   OS="linux" ;;
+  darwin)  OS="darwin" ;;
+  *)
+    echo "If you are on Windows or another unsupported OS, please follow the manual installation instructions at:"
+    echo "https://github.com/GoogleCloudPlatform/kubectl-ai#manual-installation"
+    exit 1
+    ;;
+esac
+
+# Detect ARCH
+ARCH="$(uname -m)"
+case "$ARCH" in
+  x86_64|amd64) ARCH="amd64" ;;
+  arm64|aarch64) ARCH="arm64" ;;
+  *)
+    echo "If you are on an unsupported architecture, please follow the manual installation instructions at:"
+    echo "https://github.com/GoogleCloudPlatform/kubectl-ai#manual-installation"
+    exit 1
+    ;;
+esac
+
+# Get latest version tag from GitHub API (portable, no grep -P)
+LATEST_TAG=$(curl -s "https://api.github.com/repos/$REPO/releases/latest" | sed -n 's/.*"tag_name": *"\([^"]*\)".*/\1/p')
+if [ -z "$LATEST_TAG" ]; then
+  echo "Failed to fetch latest release tag."
+  exit 1
+fi
+
+# Compose download URL
+TARBALL="kubectl-ai_${OS}_${ARCH}.tar.gz"
+URL="https://github.com/$REPO/releases/download/$LATEST_TAG/$TARBALL"
+
+# Download and extract
+echo "Downloading $URL ..."
+curl -LO "$URL"
+tar -xzf "$TARBALL"
+
+# Move binary to /usr/local/bin (may require sudo)
+echo "Installing $BINARY to /usr/local/bin (may require sudo)..."
+sudo mv "$BINARY" /usr/local/bin/
+
+# Ensure the binary is executable
+sudo chmod +x /usr/local/bin/$BINARY
```

Comment:
nit: we can make the binary executable right after untaring it to avoid using `sudo` just for that. Also, I think we don't need to do this because untaring results into an executable binary,